packages/dds/tree/src/shared-tree/treeAlpha.ts

```diff
@@ -336,6 +344,30 @@ export interface TreeAlpha {
 	 * Otherwise, this returns the key of the field that it is under (a `string`).
 	 */
 	key2(node: TreeNode): string | number | undefined;
+
+	/**
+	 * Gets the child of the given node with the given key if a child exists under that key.
+	 *
+	 * @param node - The parent node whose child is being requested.
+	 * @param key - The key under the node under which the child is being requested.
+	 *
+	 * @returns The child node or leaf value under the given key, or `undefined` if no such child exists.
+	 */
+	child(node: TreeNode, key: string | number): TreeNode | TreeLeafValue | undefined;
+
+	/**
+	 * Gets the children of the provided node, paired with their property key under the node.
+	 *
+	 * @remarks
+	 * No guarantees are made regarding the order of the children in the returned array.
```

Comment:
This should probably mention that it uses the property key, and that nodes in unknown optional fields will not be included.